library/std/src/sys/itron/thread.rs

```diff
@@ -243,8 +251,8 @@ impl Thread {
         unsafe { terminate_and_delete_task(self.task) };
 
         // In either case, we are responsible for dropping `inner`.
-        // Safety: The contents of `self.inner` will not be accessed hereafter
-        let _inner = unsafe { ManuallyDrop::take(&mut self.inner) };
+        // Safety: The contents of `*p_inner` will not be accessed hereafter
+        let _ = unsafe { Box::from_raw(self.p_inner) };
```

Comment:
Just checking, but you know that this is now going to be immediately dropped, rather than at the end of this scope, right? I'm not sure if it matters, but I feel honor-bound to ask.

Reply:
Ah, good catch! I think the intention of the original code was to guard against unwinding from a global deallocation function (although it's [not possible](https://doc.rust-lang.org/1.64.0/std/alloc/trait.GlobalAlloc.html#safety) at the moment). Will fix soon.